shimmingtoolbox/cli/b0shim.py

```diff
@@ -0,0 +1,732 @@
+from shimmingtoolbox.shim.sequencer import extend_slice, define_slices
+from shimmingtoolbox.utils import create_output_dir, set_all_loggers
+from shimmingtoolbox.shim.shim_utils import phys_to_gradient_cs
+
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+@click.group(context_settings=CONTEXT_SETTINGS,
+             help="Shim according to the specified algorithm as an argument e.g. st_shim xxxxx")
+def b0shim_cli():
+    pass
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--coil', 'coils', nargs=2, multiple=True, type=(click.Path(exists=True), click.Path(exists=True)),
+              help="Pair of filenames containing the coil profiles followed by the filename to the constraints "
+                   "e.g. --coil a.nii cons.json. If you have more than one coil, use this option more than once. "
+                   "The coil profiles and the fieldmaps (--fmap) must have matching units (if fmap is in Hz, the coil "
+                   "profiles must be in Hz/unit_shim). If you only want to shim using the scanner's gradient/shim "
+                   "coils, use the `--scanner-coil-order` option. For an example of a constraint file, "
+                   f"see: {__dir_config_scanner_constraints__}")
+@click.option('--fmap', 'fname_fmap', required=True, type=click.Path(exists=True),
+              help="Nifti filename of the B0 fieldmap. This file should contain a 3d array.")
+@click.option('--anat', 'fname_anat', type=click.Path(exists=True), required=True,
+              help="Nifti filename of the anatomical image to apply the correction onto."
+                   "This file should contain a 3d array.")
+@click.option('--mask', 'fname_mask_anat', type=click.Path(exists=True), required=False,
+              help="Nifti file used to define the spatial region to shim. This file should contain a 3d array."
+                   "The coordinate system should be the same as ``anat``'s coordinate system.")
+@click.option('--scanner-coil-order', type=click.INT, default=-1, show_default=True,
+              help="Maximum order of the shim system, allowed values are 0, 1, 2. Note that specifying 2 will return "
+                   "orders 0, 1 and 2. The 0th order is the f0 frequency.")
+@click.option('--scanner-coil-constraints', 'fname_sph_constr', type=click.Path(exists=True),
+              default=__dir_config_scanner_constraints__, show_default=True,
+              help="Constraints for the 1st and 2nd order scanner coils")
+@click.option('--slices', type=click.Choice(['interleaved', 'sequential', 'volume']), required=False,
+              default='sequential', show_default=True, help="Defines the slice ordering")
+@click.option('--slice-factor', 'slice_factor', type=click.INT, required=False, default=1, show_default=True,
+              help="Number of slices per shim for 'interleaved' and 'sequential'")
```

Comment:
```suggestion
              help="Number of slices per shimmed group. For example, if the value is '3', then with the 'sequential' mode, shimming will be performed independently on the following groups: {0,1,2}, {3,4,5}, etc. With the mode 'interleaved', it will be: {0,2,4}, {6,8,10}, etc.")
```

Reply:
possible issue: if total number of slice is not divisible by the nb of slice_factor --> issue to open-- more investigations needed